be/src/column/hash_set.h

```diff
@@ -35,6 +35,32 @@ class EqualOnSliceWithHash {
     bool operator()(const SliceWithHash& x, const SliceWithHash& y) const {
         // by comparing hash value first, we can avoid comparing real data
         // which may touch another memory area and has bad cache locality.
+        if (x.size != y.size) return false;
```

Comment:
```suggestion
        return x.size != y.size && x.hash == y.hash && memequal(x.data, x.size, y.data, y.size);
```

Reply:
I'm not so sure this will reduce some of the control dependency

Reply:
It's useless, use && can't reduce control dependency

Reply:
It seems`&&` can't reduce control dependency, but it seem it wil generate shorter assemble ... 
https://godbolt.org/z/9rE5nosqo

Reply:
Those 3 versions are almost the same 17 lines.  I'm not sure which will perform better in production. 

And I guess this is assembly code generated in an independent function.  In production, those functions will be inlined and I'm not sure the final assembly code in production.

Anyway,  I'm okay to use a compact form like `x.size == y.size && x.hash == y.hash`.

Reply:
You can try `x.size == y.size & x.hash == y.hash & memequal(x.data, x.size, y.data, y.size);`,  `&` can reduce control dependency,  more friendly to cache, but there is no short circuit, and I not sure will faster than now.